internal/services/authorization/role_management_policy.go

```diff
@@ -0,0 +1,634 @@
+		if notificationRequestorAdminAssignment, ok := existingRules["Notification_Requestor_Admin_Assignment"]; ok && notificationRequestorAdminAssignment.Values != nil {
+			if len(model.NotificationRules) == 1 {
+				if len(model.NotificationRules[0].ActiveAssignments) == 1 {
+					if len(model.NotificationRules[0].ActiveAssignments[0].AssigneeNotifications) == 1 {
+						updatedRules = append(updatedRules,
+							expandNotificationSettings(
+								notificationRequestorAdminAssignment,
+								model.NotificationRules[0].ActiveAssignments[0].AssigneeNotifications[0],
+								metadata.ResourceData.HasChange("notification_rules.0.active_assignments.0.assignee_notifications.0.additional_recipients"),
+							),
+						)
+					}
+				}
+			}
+		}
+	}
+
+	if metadata.ResourceData.HasChange("notification_rules.0.eligible_activations.0.assignee_notifications") {
+		if notificationRequestorEndUserAssignment, ok := existingRules["Notification_Requestor_EndUser_Assignment"]; ok && notificationRequestorEndUserAssignment.Values != nil {
+			if len(model.NotificationRules) == 1 {
+				if len(model.NotificationRules[0].EligibleActivations) == 1 {
+					if len(model.NotificationRules[0].EligibleActivations[0].AssigneeNotifications) == 1 {
+						updatedRules = append(updatedRules,
+							expandNotificationSettings(
+								notificationRequestorEndUserAssignment,
+								model.NotificationRules[0].EligibleActivations[0].AssigneeNotifications[0],
+								metadata.ResourceData.HasChange("notification_rules.0.eligible_activations.0.assignee_notifications.0.additional_recipients"),
+							),
+						)
+					}
+				}
+			}
+		}
+	}
+
+	return &rolemanagementpolicies.RoleManagementPolicy{
+		Id:   rolePolicy.Id,
+		Name: rolePolicy.Name,
+		Type: rolePolicy.Type,
+		Properties: &rolemanagementpolicies.RoleManagementPolicyProperties{
+			Rules: pointer.To(updatedRules),
+		},
+	}, nil
+}
+
+func expandNotificationSettings(rule rolemanagementpolicies.RawRoleManagementPolicyRuleImpl, data RoleManagementPolicyNotificationSettings, recipientChange bool) rolemanagementpolicies.RoleManagementPolicyRule {
+	var level string
+	if levelRaw, ok := rule.Values["notificationLevel"]; ok {
+		level = levelRaw.(string)
+	}
+
+	var defaultRecipients bool
+	if defaultRecipientsRaw, ok := rule.Values["isDefaultRecipientsEnabled"]; ok {
+		defaultRecipients = defaultRecipientsRaw.(bool)
+	}
+
+	var additionalRecipients []string
+	if v, ok := rule.Values["notificationRecipients"]; ok {
+		additionalRecipientsRaw := v.([]interface{})
+		additionalRecipients = make([]string, len(additionalRecipientsRaw))
+		for i, r := range additionalRecipientsRaw {
+			additionalRecipients[i] = r.(string)
+		}
+	}
+
+	if level != data.NotificationLevel {
+		level = data.NotificationLevel
+	}
+	if defaultRecipients != data.DefaultRecipients {
+		defaultRecipients = data.DefaultRecipients
+	}
+	if recipientChange {
+		additionalRecipients = data.AdditionalRecipients
```

Comment:
Are we losing recipient information with this? We're doing quite a bit of work [here](https://github.com/hashicorp/terraform-provider-azurerm/pull/25900/commits/908d91cd2048dc267c138d06a4a829dcff37cb5a#diff-9996d5f0c36b15af9ef029fd994ed24509a0b1ad5e66f1b4f40beaa0c0189d5dR560) and then overwriting it if `recipientChange` is true. Did we want to append instead of overwrite or is this correct?

Reply:
I think this is OK, on lines 560-566 we're populating with any existing additional recipients as retrieved from the policy prior to updating, and here we're selectively replacing them if the property changed in the config (it seems they'll be removed if you just don't specify them in the Update request).